### PR TITLE
Create applicationscountstore

### PR DESCRIPTION
**Pull request by sanposhiho**
**What this PR does / why we need it**:
create filestore for applications count(application insight)

**Which issue(s) this PR fixes**:

Refs: #1436 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/insight/applicationscount.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 The PipeCD Authors.
```
Comment (nghialv) [inline pkg/insight/applicationscount.go:1]: "2021"

File: pkg/insight/applicationscount.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	// Use model.ApplicationKind as the key
+	// and key "all" means data of all kind of applications.
+	Counts map[string]Count `json:"counts"`
+}
+
+type Count struct {
+	Deploying int64 `json:"deploying"`
+	Disabled  int64 `json:"disabled"`
+	Deleted   int64 `json:"deleted"`
+}
```
Comment (nghialv) [inline pkg/insight/applicationscount.go:28]: How about using a LabelSet to group the counts like this?

``` golang

type ApplicationCount struct {
  Counts []CountByLabelSet
  MinimumCreatedAt int64 // We need this field to know when the team started using our service.
  AccumulatedTo int64 `json:"accumulated_to"`
}

type ApplicationCountLabelSet struct {
   Kind string   // KUBERNETES, TERRAFORM, CLOUDRUN...
   Status string // enable, disable or deleted
}

type ApplicationCountByLabelSet struct {
   LabelSet CountLabelSet
   Count int
}
```

**Comment by nghialv** [comment]
@sanposhiho Thank you for your pull request.
I have just left some comments. Could you please check them out?

File: pkg/insight/applicationscount.go (hunk 2 above)
Comment (sanposhiho) [reply pkg/insight/applicationscount.go:28]: I think your idea is better because it's designed similarly to proto!👍

**Comment by sanposhiho** [comment]
@nghialv Thanks for your review! fixed it

**Comment by nghialv** [comment]
@sanposhiho Nice. Thank you.
I guess you are going to implement the collector part for application insight.
So I just want to say that maybe it can be done by listing all applications to build the insight data.
We don't need to query a range of unhandled ones like what we did with deployment, because compared to deployment, the number of applications is much smaller.

**Pull request merged**